crates/solver/src/settlement_submission.rs

```diff
@@ -370,6 +376,38 @@ impl SolutionSubmitter {
     }
 }
 
+/// A settlement that is ready to be used by submitters.
+#[derive(Clone, Debug)]
+pub struct SubmitterSettlement {
+    inner: Settlement,
+    encoded: EncodedSettlement,
```

Comment:
Why does the submitter suddenly has to know about both settlements?

Reply:
It uses the `EncodedSettlement` in most places, but in some places (specifically access list generation) it needs the `Settlement`. Its a much larger refactor to pull this out and I'm afraid of touching the logic too much...

Reply:
I'm still struggling to see which new code path introduced the requirement to have both versions. Didn't we also compute access lists before?
Instead of having the encoded and unencoded version maybe it's nicer to only have the unencoded version and the data needed to encode the settlement. This would just be the `Contracts` which you already pass into `SubmitterSettlement::new()`.
This seems at least slightly nicer to me because there is only a single source of truth. Whereas with the other approach you could theoretically store the encoded and unencoded version of 2 entirely different settlements.

Reply:
> I'm still struggling to see which new code path introduced the requirement to have both versions.

Both versions were always required. It is just that previously, there existed a function `f :: Settlement -> EncodedSettlement`, but now that doesn't exist anymore.

> Instead of having the encoded and unencoded version maybe it's nicer to only have the unencoded version and the data needed to encode the settlement.

Good suggestion.

Reply:
223a88a6fdef1475d294c6b19961cc0c8e6fe43b